containers-tests/tests/intmap-strictness.hs

```diff
@@ -76,6 +82,76 @@ pInsertLookupWithKeyValueStrict f k v m
                      not (isBottom $ M.insertLookupWithKey (const3 1) k bottom m)
     | otherwise    = isBottom $ M.insertLookupWithKey (apply3 f) k bottom m
 
+------------------------------------------------------------------------
+-- test a corner case of fromAscList: if the list contains duplicate
+-- keys, then (only) the first of the given values is not evaluated.
```

Comment:
That seems like a surprising property to me. I think https://github.com/haskell/containers/issues/473 tracks what we could about it. Could you add a reference to that issue?

Reply:
done.